kong/plugins/rate-limiting/consumer_api.lua

```diff
@@ -0,0 +1,56 @@
+local crud = require "kong.api.crud_helpers"
+local timestamp = require "kong.tools.timestamp"
+local constants = require "kong.constants"
+local rate_limit_utils = require "kong.plugins.rate-limiting.utils"
+
+local string_format = string.format
+
+local function format_usage(usage)
+  local usage_for_field
+  local response_data = { rate = {} }
+  local no_limit_value = constants.RATELIMIT.USAGE.NO_LIMIT_VALUE
+
+  for k, field_name in ipairs(constants.RATELIMIT.PERIODS) do
+    usage_for_field = usage[field_name] or {}
+    response_data.rate[string_format("limit-%s", field_name)] = usage_for_field.limit or no_limit_value
+    response_data.rate[string_format("remaining-%s", field_name)] = usage_for_field.remaining or no_limit_value
+  end
+
+  return response_data
+end
+
+return {
+
+  ["/apis/:name_or_id/plugins/rate-limiting/usage/:key"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_consumer_by_username_or_id(self, dao_factory, helpers)
+      self.params.consumer_id = self.consumer.id
+
+      local data, err = dao_factory.keyauth_credentials:find_by_keys({ key = self.params.key })
```

Comment:
I think consumer id is fine for a start.

does the "custom_id" is unique for all consumers ? 
if so, it may fit better here - its a user defined, and we don't hand internal id for api.